libr/core/pseudo.c

```diff
@@ -1,7 +1,148 @@
 /* radare - LGPL - Copyright 2015-2016 - pancake */
 
 #include <r_core.h>
+#define TYPE_NONE		0
+#define TYPE_STR		1
+#define TYPE_SYM		2
+#define IS_CHAR(x)		(IS_UPPER(x) || IS_LOWER(x))
+#define IS_STRING(x)	(x+3<end && *x == 's' && *(x+1) == 't' && *(x+2) == 'r' && *(x+3) == '.')
+#define IS_SYMBOL(x)	(x+3<end && *x == 's' && *(x+1) == 'y' && *(x+2) == 'm' && *(x+3) == '.')
 
+static void find_and_change (char* in, int len) {
+	char *comment, *left, *right, *linebegin, *end;
+	int leftlen, rightlen, leftpos, leftcolor, commentcolor, rightcolor, linecount;
+	int type;
+	if (!in || len <= 0) {
+		return;
+	}
+	end = in + len;
+
+	comment = NULL;
+	// >>
+	right = NULL;
+	// <<
+	left = NULL;
+	leftlen = 0;
+	rightlen = 0;
+	linecount = 0;
+	leftpos = 0;
+	leftcolor = 0;
+	rightcolor = 0;
+	commentcolor = 0;
+	type = TYPE_NONE;
+	linebegin = in + 1;
+	for (linebegin = in; in < end; ++in) {
+		if (*in == '\n' || !*in) {
+			if (type == TYPE_SYM && linecount < 1) {
+				linecount++;
+				linebegin = in + 1;
+				continue;
+			}
+			if (type != TYPE_NONE && right && left && rightlen > 0 && leftlen > 0) {
+				char* copy = NULL;
+				if (leftlen > rightlen && (copy = (char*) malloc (leftlen)) != NULL) {
```

Comment:
put the malloc inside the conditional blocknot inside the conditional because in case of fail it will fall thrugh the next condition which may be unexpected